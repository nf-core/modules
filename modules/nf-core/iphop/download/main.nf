process IPHOP_DOWNLOAD {
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/iphop:1.3.2--pyhdfd78af_0':
        'biocontainers/iphop:1.3.2--pyhdfd78af_0' }"

    output:
    path "iphop_db/"        , emit: iphop_db
    path "versions.yml"     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    mkdir -p download_dir
    mkdir -p iphop_db

    iphop \\
        download \\
        --db_dir download_dir \\
        --no_prompt \\
        $args

    rm download_dir/*.tar.*
    mv download_dir/*/* iphop_db

    iphop \\
        download \\
        --db_dir iphop_db \\
        --no_prompt \\
        --full_verify \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        iphop: \$(echo \$(iphop --version 2>&1) | head -n 1 | sed 's/^.*iPHoP v//; s/: integrating.*\$//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    mkdir -p iphop_db/db/
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.fna
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.ndb
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.nhr
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.nin
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.not
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.nsq
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.ntf
    touch iphop_db/db/All_CRISPR_spacers_nr_clean.nto
    touch iphop_db/db/GTDBtkr202_and_newrepr_s2_mat.pkl
    mkdir -p iphop_db/db/Host_Genomes
    touch iphop_db/db/Host_Genomes/Host_Genomes.ndb
    touch iphop_db/db/Host_Genomes/Host_Genomes.nhr
    touch iphop_db/db/Host_Genomes/Host_Genomes.nin
    touch iphop_db/db/Host_Genomes/Host_Genomes.not
    touch iphop_db/db/Host_Genomes/Host_Genomes.nsq
    touch iphop_db/db/Host_Genomes/Host_Genomes.ntf
    touch iphop_db/db/Host_Genomes/Host_Genomes.nto
    mkdir -p iphop_db/db/php_db
    mkdir -p iphop_db/db/rafah_data
    touch iphop_db/db/rafah_data/HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3f
    touch iphop_db/db/rafah_data/HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3i
    touch iphop_db/db/rafah_data/HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3m
    touch iphop_db/db/rafah_data/HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3p
    touch iphop_db/db/rafah_data/HP_Ranger_Model_3_Valid_Cols.txt
    touch iphop_db/db/rafah_data/MMSeqs_Clusters_Ranger_Model_1+2+3_Clean.RData
    touch iphop_db/db/rafah_data/RaFAH_ref_cds.count.tsv
    touch iphop_db/db/rafah_data/RaFAH_ref_cds.dmnd
    mkdir -p iphop_db/db/rewish_models
    touch iphop_db/db/rewish_models/Batch_1.pkl
    mkdir -p iphop_db/db/wish_data/Decoy_db
    touch iphop_db/db/wish_data/Decoy_db/Decoy_phages.fna
    mkdir -p iphop_db/db_infos
    touch iphop_db/db_infos/All_CRISPR_array_size.tsv
    touch iphop_db/db_infos/All_CRISPR_spacers_nr_clean.metrics.csv
    touch iphop_db/db_infos/Host_Genomes.tsv
    touch iphop_db/db_infos/List_contigs_removed_blast.tsv
    touch iphop_db/db_infos/Translate_genus_to_full_taxo.tsv
    touch iphop_db/db_infos/Wish_negFits.csv
    touch iphop_db/db_infos/gtdbtk.ar122.decorated.tree
    touch iphop_db/db_infos/gtdbtk.bac120.decorated.tree
    touch iphop_db/md5checkfile.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        iphop: \$(echo \$(iphop --version 2>&1) | head -n 1 | sed 's/^.*iPHoP v//; s/: integrating.*\$//' ))
    END_VERSIONS
    """
}
