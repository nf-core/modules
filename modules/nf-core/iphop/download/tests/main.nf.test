nextflow_process {

    name "Test Process IPHOP_DOWNLOAD"
    config "./nextflow.config"
    script "../main.nf"
    process "IPHOP_DOWNLOAD"

    tag "modules"
    tag "modules_nfcore"
    tag "iphop"
    tag "iphop/download"

    /* The test database is 4.5Gb and takes a long time to download
    test("iPHoP_db_rw_for-test") {
        tag "test"
        when {
            params {
                module_args = '--db_version iPHoP_db_rw_for-test'
            }
            process {
                """
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    getAllFilesFromDir(process.out.iphop_db[0], true, null, null, ['*', '**//*'])*.name // Replace '//' by '/'
                ).match() }
            )
        }
    }
    */

    test("iPHoP_db_rw_for-test -- stub") {
        tag "stub"
        options '-stub'
        when {
            params {
                module_args = '--db_version iPHoP_db_rw_for-test'
            }
            process {
                """
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.iphop_db,
                    process.out.versions,
                    path(process.out.versions[0]).yaml
                ).match() }
            )
        }
    }
}
