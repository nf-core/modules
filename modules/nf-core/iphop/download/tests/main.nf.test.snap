{
    "iPHoP_db_rw_for-test": {
        "content": [
            [
                "versions.yml:md5,fb282fb8b508907399b215a764f65aff"
            ],
            [
                "iphop_db",
                "db",
                "All_CRISPR_spacers_nr_clean.fna",
                "All_CRISPR_spacers_nr_clean.ndb",
                "All_CRISPR_spacers_nr_clean.nhr",
                "All_CRISPR_spacers_nr_clean.nin",
                "All_CRISPR_spacers_nr_clean.not",
                "All_CRISPR_spacers_nr_clean.nsq",
                "All_CRISPR_spacers_nr_clean.ntf",
                "All_CRISPR_spacers_nr_clean.nto",
                "GTDBtkr202_and_newrepr_s2_mat.pkl",
                "Host_Genomes",
                "Host_Genomes.ndb",
                "Host_Genomes.nhr",
                "Host_Genomes.nin",
                "Host_Genomes.not",
                "Host_Genomes.nsq",
                "Host_Genomes.ntf",
                "Host_Genomes.nto",
                "php_db",
                "rafah_data",
                "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3f",
                "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3i",
                "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3m",
                "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3p",
                "HP_Ranger_Model_3_Valid_Cols.txt",
                "MMSeqs_Clusters_Ranger_Model_1+2+3_Clean.RData",
                "RaFAH_ref_cds.count.tsv",
                "RaFAH_ref_cds.dmnd",
                "rewish_models",
                "Batch_1.pkl",
                "wish_data",
                "Decoy_db",
                "Decoy_phages.fna",
                "db_infos",
                "All_CRISPR_array_size.tsv",
                "All_CRISPR_spacers_nr_clean.metrics.csv",
                "Host_Genomes.tsv",
                "List_contigs_removed_blast.tsv",
                "Translate_genus_to_full_taxo.tsv",
                "Wish_negFits.csv",
                "gtdbtk.ar122.decorated.tree",
                "gtdbtk.bac120.decorated.tree",
                "md5checkfile.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T13:55:12.464298703"
    },
    "iPHoP_db_rw_for-test -- stub": {
        "content": [
            [
                [
                    [
                        "All_CRISPR_spacers_nr_clean.fna:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.nto:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "GTDBtkr202_and_newrepr_s2_mat.pkl:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "Host_Genomes.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Host_Genomes.nto:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        "php_db:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3f:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3i:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3m:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "HP_Ranger_Model_3_Filtered_0.9_Valids.hmm.h3p:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "HP_Ranger_Model_3_Valid_Cols.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "MMSeqs_Clusters_Ranger_Model_1+2+3_Clean.RData:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "RaFAH_ref_cds.count.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "RaFAH_ref_cds.dmnd:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        [
                            "Batch_1.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        [
                            [
                                "Decoy_phages.fna:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ]
                        ]
                    ],
                    [
                        "All_CRISPR_array_size.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "All_CRISPR_spacers_nr_clean.metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "Host_Genomes.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "List_contigs_removed_blast.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "Translate_genus_to_full_taxo.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "Wish_negFits.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "gtdbtk.ar122.decorated.tree:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "gtdbtk.bac120.decorated.tree:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    "md5checkfile.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,fb282fb8b508907399b215a764f65aff"
            ],
            {
                "IPHOP_DOWNLOAD": {
                    "iphop": "1.3.2"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T19:49:59.824775487"
    }
}